mlir/lib/Bindings/Python/IRAttributes.cpp

```diff
@@ -1494,14 +1501,56 @@ class PyDenseResourceElementsAttribute
     return PyDenseResourceElementsAttribute(contextWrapper->getRef(), attr);
   }
 
+  static PyDenseResourceElementsAttribute
+  getFromBufferNdarray(const nb::ndarray<nb::any_contig>& buffer, const std::string &name, const PyType &type,
+                std::optional<size_t> alignment, bool isMutable,
+                DefaultingPyMlirContext contextWrapper) {
+    if (!mlirTypeIsAShaped(type)) {
+      throw std::invalid_argument(
+          "Constructing a DenseResourceElementsAttr requires a ShapedType.");
+    }
+
+    // Do not request any conversions as we must ensure to use caller
+    // managed memory.
+    if (!buffer.is_valid()) {
+      throw nb::python_error();
+    }
+
+    // Infer alignment to be the stride of one element if not explicit.
+    size_t inferredAlignment;
+    if (alignment)
+      inferredAlignment = *alignment;
+    else
+      inferredAlignment = buffer.stride_ptr()[buffer.ndim() - 1];
+
+    size_t rawBufferSize = buffer.size();
+    MlirAttribute attr = mlirUnmanagedDenseResourceElementsAttrGet(
+        type, toMlirStringRef(name), buffer.data(), rawBufferSize,
+        inferredAlignment, isMutable, NULL, NULL);
+    if (mlirAttributeIsNull(attr)) {
+      throw std::invalid_argument(
+          "DenseResourceElementsAttr could not be constructed from the given "
+          "buffer. "
+          "This may mean that the Python buffer layout does not match that "
+          "MLIR expected layout and is a bug.");
+    }
+    return PyDenseResourceElementsAttribute(contextWrapper->getRef(), attr);
+  }
+
   static void bindDerived(ClassTy &c) {
     c.def_static(
         "get_from_buffer", PyDenseResourceElementsAttribute::getFromBuffer,
         nb::arg("array"), nb::arg("name"), nb::arg("type"),
         nb::arg("alignment").none() = nb::none(), nb::arg("is_mutable") = false,
         nb::arg("context").none() = nb::none(),
         kDenseResourceElementsAttrGetFromBufferDocstring);
-  }
+    c.def_static(
+        "get_from_buffer_ndarray", PyDenseResourceElementsAttribute::getFromBufferNdarray,
```

Comment:
nit naming: 
```suggestion
        "get_from_ndarray", PyDenseResourceElementsAttribute::getFromNdarray,
```
Just to be clear with are not strictly dealing with Python Buffers but framework N-Dimensional Arrays.